sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample10_MockingReadingEventsTests.cs

```diff
@@ -0,0 +1,290 @@
+                EventPosition startingPosition = EventPosition.Earliest;
+
+                await foreach (PartitionEvent partitionEvent in consumer.ReadEventsFromPartitionAsync(
+                    firstPartition,
+                    startingPosition,
+                    cancellationSource.Token))
+                {
+                    string readFromPartition = partitionEvent.Partition.PartitionId;
+                    ReadOnlyMemory<byte> eventBodyBytes = partitionEvent.Data.EventBody.ToMemory();
+
+                    Debug.WriteLine($"Read event of length { eventBodyBytes.Length } from { readFromPartition }");
+                }
+            }
+            catch (TaskCanceledException)
+            {
+                // This is expected if the cancellation token is
+                // signaled.
+            }
+            finally
+            {
+                await consumer.CloseAsync();
+            }
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic unit test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ReadPartitionWithReceiver()
+        {
+            #region Snippet:EventHubs_Sample10_MockingReadPartitionWithReceiver
+
+            var mockEventHubProducerClient = new Mock<EventHubProducerClient>();
+            mockEventHubProducerClient
+                .Setup(c => c.GetPartitionIdsAsync(It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new string[1] { "sample id" });
+
+            var mockEventHubPartitionReceiver = new Mock<PartitionReceiver>();
+            var mockBatchData = GenerateBatchData();
+            mockEventHubPartitionReceiver
+                .Setup(r => r.ReceiveBatchAsync(It.IsAny<int>(), It.IsAny<TimeSpan>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(mockBatchData)
+                .Verifiable();
+
+            using CancellationTokenSource cancellationSource = new CancellationTokenSource();
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+            string firstPartition;
+
+            await using (var producer = mockEventHubProducerClient.Object)
+            {
+                firstPartition = (await producer.GetPartitionIdsAsync()).First();
+            }
+
+            var receiver = mockEventHubPartitionReceiver.Object;
+            try
+            {
+                while (!cancellationSource.IsCancellationRequested)
+                {
+                    int batchSize = 50;
+                    TimeSpan waitTime = TimeSpan.FromSeconds(1);
+
+                    IEnumerable<EventData> eventBatch = await receiver.ReceiveBatchAsync(
+                        batchSize,
+                        waitTime,
+                        cancellationSource.Token);
+
+                    //do we need assertion like this?
+                    for (var index = 0; index < mockBatchData.Count(); ++index)
```

Comment:
We probably don't need the assert; a comment would cover it well enough, I think.